apps/portal/app/routes/app+/quest+/index.tsx

```diff
@@ -38,19 +36,114 @@ import {
 } from 'app/consts'
 import { isAddress } from 'viem'
 
+interface GraphQLResponse<T> {
+  data?: T
+  errors?: Array<{ message: string }>
+}
+
+const GetMintCountDocument = {
+  query: `
+    query GetMintCountUntilDate($address: String!, $cutoff_timestamp: Int!) {
+      voucherRedeemedEvents(
+        where: { redeemer: $address, timestamp_lte: $cutoff_timestamp }
+      ) {
+        totalCount
+      }
+    }
+  `,
+} as const
+
+interface GetMintCountQuery {
+  voucherRedeemedEvents: {
+    totalCount: number
+  }
+}
+
+interface GetMintCountQueryVariables {
+  address: string
+  cutoff_timestamp: number
+}
+
+async function fetchGraphQL<T, V>(
+  document: { query: string },
+  variables: V,
+): Promise<GraphQLResponse<T>> {
+  const endpoint = process.env.RELIC_GRAPHQL_ENDPOINT
```

Comment:
Nice job! We'll need to also add this to the build action scripts and Dockerfile. Likely makes sense to do in this same PR.